sys/include/net/conn.h

```diff
@@ -12,19 +12,55 @@
  * @brief       Provides a minimal common API for applications to connect to the
  *              different network stacks.
  *
- * This module provides a minimal common API for applications to connect over different network
- * stacks. For each network stack there is supposed to be at least one connection type
- * implementation. Note that this definition gives no restriction on how a connection type should be
- * structured for simplicity and modularity. As a result, it can't give any guarantee that an
- * implementation keeps them compatible to each other. For example, an implementation might allow,
- * that a UDP receive function is called on a raw IPv6 connection object and even choose to do so
- * for valid reasons (e.g. code size), but this definition does not impose this on the
- * implementation. Currently there are the following option types defined:
+ * About
+ * =====
+ *
+* +---------------+
+ * |  Application  |
+ * +---------------+
+ *         ^
+ *         |
+ *         v
+ *       conn
+ *         ^
+ *         |
+ *         v
+ * +---------------+
+ * | Network Stack |
+ * +---------------+
+ *
+ * This module provides a minimal set of functions to establish a connection using
+ * different types of connections. Together, they serve as a common API
+ * that connects application- and network stack code.
+ *
+ * Currently the following connection types are defined:
  *
  * * @ref conn_ip_t (net/conn/ip.h): raw IP connections
  * * @ref conn_tcp_t (net/conn/tcp.h): TCP connections
  * * @ref conn_udp_t (net/conn/udp.h): UDP connections
  *
+ * Each network stack must implement at least one connection type.
+ *
+ * Note that this definition gives no restriction on how a connection type should be
+ * structured for simplicity and modularity. As a result, it can't give any guarantee that an
+ * implementation keeps them compatible to each other. For example, an implementation might allow
+ * a UDP receive function to be called on a raw IPv6 connection object and even choose to do so
+ * for valid reasons (e.g. code size), but this definition does not impose this on the
+ * implementation.
```

Comment:
That's from the old documentation, I didn't write it ^^ I understood it as a warning that you can't rely on the implementations being completely interchangeable (for example, their definition of conn_udp may be different)?